codalab/machines/remote_machine.py

```diff
@@ -196,7 +196,9 @@ def get_field(path, col):
             # Attach all GPUs if any
             f.write('devices=$(/bin/ls /dev/nvidia* 2>/dev/null)\n')
             f.write('if [ -n "$devices" ]; then devices=$(for d in $devices; do echo --device $d:$d; done); fi\n')
-            resource_args += ' $devices'
+            f.write('libcuda=$(/sbin/ldconfig -p 2>/dev/null | grep "libcuda.so$" | grep "x86-64" | head -n 1 | cut -d " " -f 4)\n')
+            f.write('if [ -n "$libcuda" ]; then libcuda="-v $libcuda:/usr/lib/x86_64-linux-gnu/libcuda.so"; fi\n')
```

Comment:
I see...could you just write a comment that this works only with 64-bit?

Otherwise, LGTM

On Sun, Jan 31, 2016 at 4:32 PM, Konstantin Lopyrev <
notifications@github.com> wrote:

> In codalab/machines/remote_machine.py
> <https://github.com/codalab/codalab-cli/pull/324#discussion_r51370993>:
>
> > @@ -196,7 +196,9 @@ def get_field(path, col):
> >              # Attach all GPUs if any
> >              f.write('devices=$(/bin/ls /dev/nvidia* 2>/dev/null)\n')
> >              f.write('if [ -n "$devices" ]; then devices=$(for d in $devices; do echo --device $d:$d; done); fi\n')
> > -            resource_args += ' $devices'
> > +            f.write('libcuda=$(/sbin/ldconfig -p 2>/dev/null | grep "libcuda.so$" | grep "x86-64" | head -n 1 | cut -d " " -f 4)\n')
> > +            f.write('if [ -n "$libcuda" ]; then libcuda="-v $libcuda:/usr/lib/x86_64-linux-gnu/libcuda.so"; fi\n')
>
> If I do just $libcuda:$libcuda for some machines it would put the library
> into a directory that's not on the linker search path for the Tensorflow
> docker image. For example, putting it in /usr/lib64 doesn't work with that
> image.
>
> If you don't ensure that the 64 bit version of the library is used, you
> could instead get the 32 bit version. That happens on my machine. When you
> try to use the 32 bit version I get the error: dlerror: libcuda.so: wrong
> ELF class: ELFCLASS32
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/codalab/codalab-cli/pull/324/files#r51370993>.
>